src/test/regress/specs/isolation_add_node_vs_copy_reference_table.spec

```diff
@@ -0,0 +1,69 @@
+# the test expects to have zero nodes in pg_dist_node at the beginning
+# add single one of the nodes for the purpose of the test
+setup
+{	
+	SELECT nodename, nodeport, isactive FROM master_add_node('localhost', 57637);
+	
+    CREATE TABLE test_reference_table (test_id integer);
+    SELECT create_reference_table('test_reference_table');
+}
+
+# ensure that both nodes exists for the remaining of the isolation tests
+teardown
+{
+	DROP TABLE test_reference_table;
+	SELECT master_remove_node(nodename, nodeport) FROM pg_dist_node;
+}
+
+session "s1"
+
+step "s1-begin"
+{
+    BEGIN;
+}
+
+step "s1-add-second-worker"
+{
+	SELECT nodename, nodeport, isactive FROM master_add_node('localhost', 57638);
+}
+
+step "s1-remove-second-worker"
+{
+	SELECT master_remove_node('localhost', 57638);
+}
+
+step "s1-commit"
+{
+    COMMIT;
+}
+
+session "s2"
+
+# COPY accesses all shard/placement metadata, so should be enough for
+# loading the cache
+step "s2-load-metadata-cache"
```

Comment:
Not strictly necessary, but seemed better to give a separate explicit name. Do you prefer using the same step for both?